diff --git a/platformtheme/CMakeLists.txt b/platformtheme/CMakeLists.txt
index 266a9c7..e340411 100644
--- a/platformtheme/CMakeLists.txt
+++ b/platformtheme/CMakeLists.txt
@@ -26,18 +26,20 @@ target_link_libraries(cutecosmictheme PRIVATE Qt::GuiPrivate Qt::QuickControls2
 # Find out where to install the plugin
 find_package(Qt6 COMPONENTS CoreTools QUIET CONFIG)
 
-if(NOT TARGET Qt6::qtpaths)
-    message(SEND_ERROR "Could not find qtpaths tool, can't install the plugin!")
-else()
-    get_target_property(qtpaths_executable Qt6::qtpaths LOCATION)
-    execute_process(
-        COMMAND ${qtpaths_executable} -qt-query QT_INSTALL_PLUGINS
-        OUTPUT_VARIABLE QT_INSTALL_PLUGINS
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-
-    message(STATUS "Installing plugin to ${QT_INSTALL_PLUGINS}")
-
-    set_target_properties(cutecosmictheme PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)
-    install(TARGETS cutecosmictheme DESTINATION ${QT_INSTALL_PLUGINS}/platformthemes)
+if(NOT QT_INSTALL_PLUGINS)
+  if(NOT TARGET Qt6::qtpaths)
+      message(SEND_ERROR "Could not find qtpaths tool, can't install the plugin!")
+  else()
+      get_target_property(qtpaths_executable Qt6::qtpaths LOCATION)
+      execute_process(
+          COMMAND ${qtpaths_executable} -qt-query QT_INSTALL_PLUGINS
+          OUTPUT_VARIABLE QT_INSTALL_PLUGINS
+          OUTPUT_STRIP_TRAILING_WHITESPACE
+      )
+  endif()
 endif()
+
+message(STATUS "Installing plugin to ${QT_INSTALL_PLUGINS}")
+
+set_target_properties(cutecosmictheme PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)
+install(TARGETS cutecosmictheme DESTINATION ${QT_INSTALL_PLUGINS}/platformthemes)
\ No newline at end of file