diff --git a/steamos-manager/src/session.rs b/steamos-manager/src/session.rs
index 1754f46..642c817 100644
--- a/steamos-manager/src/session.rs
+++ b/steamos-manager/src/session.rs
@@ -32,7 +32,7 @@ use crate::manager::root::RootManagerProxy;
 use crate::systemd::{ActiveState, JobMode, SystemdUnit};
 use crate::{Service, path};

-const CONFIG_PREFIX_USR: &str = "/usr/lib/sddm/sddm.conf.d";
+const CONFIG_PREFIX_USR: &str = "/etc/sddm.conf.d";
 const CONFIG_PREFIX: &str = "/etc/sddm.conf.d";
 const SESSION_CHECK_PATH_LEGACY: &str = "steamos.conf";
 const CONFIG_PATH_LEGACY: &str = "zz-steamos-autologin.conf";