diff --git a/sycl/cmake/modules/AddSYCLUnitTest.cmake b/sycl/cmake/modules/AddSYCLUnitTest.cmake
index a93896ecce354..b4b01b4fb45ea 100644
--- a/sycl/cmake/modules/AddSYCLUnitTest.cmake
+++ b/sycl/cmake/modules/AddSYCLUnitTest.cmake
@@ -121,6 +121,7 @@ function(add_sycl_unittest_internal test_dirname link_variant is_preview)
       mockOpenCL
       LLVMTestingSupport
       OpenCL-Headers
+      emhash::emhash
       unified-runtime::mock
       ${SYCL_LINK_LIBS}
     )
# This was renamed to BuildUnifiedRuntime later
diff --git a/sycl/cmake/modules/FetchUnifiedRuntime.cmake b/sycl/cmake/modules/FetchUnifiedRuntime.cmake
index 3196202dda660..e90eb7887d14d 100644
--- a/sycl/cmake/modules/FetchUnifiedRuntime.cmake
+++ b/sycl/cmake/modules/FetchUnifiedRuntime.cmake
@@ -361,7 +361,6 @@ if(CMAKE_SYSTEM_NAME STREQUAL Windows)
       -DUMF_BUILD_SHARED_LIBRARY:BOOL=${UMF_BUILD_SHARED_LIBRARY}
       -DUMF_LINK_HWLOC_STATICALLY:BOOL=${UMF_LINK_HWLOC_STATICALLY}
       -DUMF_DISABLE_HWLOC:BOOL=${UMF_DISABLE_HWLOC}
-      -DSYCL_EMHASH_DIR:STRING=${SYCL_EMHASH_DIR}
       # Enable d suffix in UMF
       -DUMF_USE_DEBUG_POSTFIX:BOOL=ON
   )
diff --git a/sycl/cmake/modules/FetchEmhash.cmake b/sycl/cmake/modules/FetchEmhash.cmake
index a87f799732d6b..b5fae1dafa254 100644
--- a/sycl/cmake/modules/FetchEmhash.cmake
+++ b/sycl/cmake/modules/FetchEmhash.cmake
@@ -1,24 +1,21 @@
 # Finds or fetches emhash.
-if(DEFINED SYCL_EMHASH_DIR OR DEFINED EMHASH_SYS_LOC)
+if(TARGET emhash::emhash)
   return()
 endif()
-find_file(EMHASH_SYS_LOC "hash_table8.hpp" PATH_SUFFIXES "emhash")
-if(NOT EMHASH_SYS_LOC)
+find_package(emhash QUIET)
+if(NOT emhash_FOUND)
   set(EMHASH_REPO https://github.com/ktprime/emhash)
   message(STATUS "Will fetch emhash from ${EMHASH_REPO}")
   FetchContent_Declare(emhash
     GIT_REPOSITORY    ${EMHASH_REPO}
-    GIT_TAG           3ba9abdfdc2e0430fcc2fd8993cad31945b6a02b
+    GIT_TAG           09fb0439de0c65e387498232c89d561ee95fbd60
     SOURCE_SUBDIR  emhash
   )
   FetchContent_MakeAvailable(emhash)
 
-  # FetchContent downloads the files into a directory with
-  # '-src' as the suffix and emhash has the headers in the
-  # top level directory in the repo, so copy the headers to a directory
-  # named `emhash` so source files can include with <emhash/header.hpp>
-  file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/include/emhash)
-  file(GLOB HEADERS "${emhash_SOURCE_DIR}/*.h*")
-  file(COPY ${HEADERS} DESTINATION ${CMAKE_BINARY_DIR}/include/emhash)
-  set(SYCL_EMHASH_DIR ${CMAKE_BINARY_DIR}/include/ CACHE INTERNAL "")
+  # The official cmake install target uses the 'emhash' namespace,
+  # so emulate that here so client code can use a single target name for both cases.
+  add_library(emhash INTERFACE)
+  target_include_directories(emhash SYSTEM INTERFACE ${emhash_SOURCE_DIR})
+  add_library(emhash::emhash ALIAS emhash)
 endif()
diff --git a/sycl/source/CMakeLists.txt b/sycl/source/CMakeLists.txt
index 38ffd232fcbbe..bb11518b003b0 100644
--- a/sycl/source/CMakeLists.txt
+++ b/sycl/source/CMakeLists.txt
@@ -115,14 +115,7 @@ function(add_sycl_rt_library LIB_NAME LIB_OBJ_NAME)
     target_include_directories(${LIB_OBJ_NAME} PRIVATE ${zstd_INCLUDE_DIR})
   endif()
 
-  if (EMHASH_SYS_LOC)
-    # If we are using system emhash, include the system directory
-    # containing the emhash folder and note this location so
-    # other modules using emhash can find it.
-    cmake_path(GET EMHASH_SYS_LOC PARENT_PATH EMHASH_SYS_LOC)
-    set(SYCL_EMHASH_DIR ${EMHASH_SYS_LOC} CACHE INTERNAL "")
-    target_include_directories(${LIB_OBJ_NAME} PRIVATE ${EMHASH_SYS_LOC})
-  endif()
+  target_link_libraries(${LIB_OBJ_NAME} PRIVATE emhash::emhash)
 
   # ur_win_proxy_loader
   if (WIN32)
diff --git a/sycl/source/detail/unordered_multimap.hpp b/sycl/source/detail/unordered_multimap.hpp
index 4d495ed065407..e96d718708d17 100644
--- a/sycl/source/detail/unordered_multimap.hpp
+++ b/sycl/source/detail/unordered_multimap.hpp
@@ -7,7 +7,7 @@
 //===--------------------------------------------------------------===//
 #pragma once
 #include <detail/hashers.hpp>
-#include <emhash/hash_table8.hpp>
+#include <hash_table8.hpp>
 
 #include <utility>
 #include <vector>
diff --git a/sycl/test/CMakeLists.txt b/sycl/test/CMakeLists.txt
index 74d34fc816b81..4e7607e749319 100644
--- a/sycl/test/CMakeLists.txt
+++ b/sycl/test/CMakeLists.txt
@@ -20,6 +20,8 @@ if(SYCL_ENABLE_EXTENSION_JIT)
   list(APPEND test_includes ${LLVM_EXTERNAL_SYCL_JIT_SOURCE_DIR}/jit-compiler/include)
   list(APPEND test_includes ${LLVM_EXTERNAL_SYCL_JIT_SOURCE_DIR}/common/include)
 endif(SYCL_ENABLE_EXTENSION_JIT)
+get_target_property(EMHASH_INCLUDES emhash::emhash INTERFACE_INCLUDE_DIRECTORIES)
+list(APPEND test_includes ${EMHASH_INCLUDES})
 list(JOIN test_includes ":" TEST_INCLUDE_PATH)
 
 configure_lit_site_cfg(
diff --git a/xptifw/include/xpti_string_table.hpp b/xptifw/include/xpti_string_table.hpp
index bd48df01921b1..5106e7eb85264 100644
--- a/xptifw/include/xpti_string_table.hpp
+++ b/xptifw/include/xpti_string_table.hpp
@@ -7,7 +7,7 @@
 
 #include "parallel_hashmap/phmap.h"
 #include "xpti/xpti_data_types.h"
-#include <emhash/hash_table7.hpp>
+#include <hash_table7.hpp>
 
 #include <atomic>
 #include <shared_mutex>
diff --git a/xptifw/src/CMakeLists.txt b/xptifw/src/CMakeLists.txt
index d3abd2944845f..07232c922725c 100644
--- a/xptifw/src/CMakeLists.txt
+++ b/xptifw/src/CMakeLists.txt
@@ -3,7 +3,7 @@ include(FetchContent)
 
 # The sycl target should download or find emhash but may not
 # for standalone builds.
-if (NOT SYCL_EMHASH_DIR)
+if (NOT TARGET emhash::emhash)
   list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/../../sycl/cmake/modules")
   include(FetchEmhash)
 endif()
@@ -37,11 +37,12 @@ function(add_xpti_library LIB_NAME)
   target_compile_definitions(${LIB_NAME} PRIVATE -DXPTI_API_EXPORTS)
   target_include_directories(${LIB_NAME} PUBLIC
     $<BUILD_INTERFACE:${XPTIFW_DIR}/include>
-    $<BUILD_INTERFACE:${SYCL_EMHASH_DIR}>
     $<BUILD_INTERFACE:${XPTIFW_PARALLEL_HASHMAP_HEADERS}>
     $<BUILD_INTERFACE:${XPTI_DIR}/include>
   )
 
+  target_link_libraries(${LIB_NAME} PUBLIC emhash::emhash)
+
   find_package(Threads REQUIRED)
   target_link_libraries(${LIB_NAME} PUBLIC ${CMAKE_DL_LIBS} Threads::Threads)
 
diff --git a/xptifw/src/xpti_trace_framework.cpp b/xptifw/src/xpti_trace_framework.cpp
index 2c9db92caa669..2845849e7bc99 100644
--- a/xptifw/src/xpti_trace_framework.cpp
+++ b/xptifw/src/xpti_trace_framework.cpp
@@ -19,7 +19,7 @@
 #include "xpti_int64_hash_table.hpp"
 #include "xpti_object_table.hpp"
 #include "xpti_string_table.hpp"
-#include <emhash/hash_table7.hpp>
+#include <hash_table7.hpp>
 
 #include <algorithm>
 #include <array>