diff --git a/xptifw/src/CMakeLists.txt b/xptifw/src/CMakeLists.txt
index de2274129582..e4de18942209 100644
--- a/xptifw/src/CMakeLists.txt
+++ b/xptifw/src/CMakeLists.txt
@@ -8,18 +8,20 @@ if (NOT SYCL_EMHASH_DIR)
   include(FetchEmhash)
 endif()

-set(PARALLEL_HASHMAP_REPO https://github.com/greg7mdp/parallel-hashmap.git)
-message(STATUS "Will fetch parallel-hashmap from ${PARALLEL_HASHMAP_REPO}")
-FetchContent_Declare(parallel-hashmap
-  GIT_REPOSITORY    ${PARALLEL_HASHMAP_REPO}
-  GIT_TAG           8a889d3699b3c09ade435641fb034427f3fd12b6
-)
-
-FetchContent_GetProperties(parallel-hashmap)
-FetchContent_MakeAvailable(parallel-hashmap)
+if (NOT DEFINED XPTIFW_PARALLEL_HASHMAP_HEADERS)
+  set(PARALLEL_HASHMAP_REPO https://github.com/greg7mdp/parallel-hashmap.git)
+  message(STATUS "Will fetch parallel-hashmap from ${PARALLEL_HASHMAP_REPO}")
+  FetchContent_Declare(parallel-hashmap
+    GIT_REPOSITORY    ${PARALLEL_HASHMAP_REPO}
+    GIT_TAG           8a889d3699b3c09ade435641fb034427f3fd12b6
+  )

-set(XPTIFW_PARALLEL_HASHMAP_HEADERS "${parallel-hashmap_SOURCE_DIR}")
+  FetchContent_GetProperties(parallel-hashmap)
+  FetchContent_MakeAvailable(parallel-hashmap)

+  set(XPTIFW_PARALLEL_HASHMAP_HEADERS "${parallel-hashmap_SOURCE_DIR}")
+endif()
+
 file(GLOB SOURCES *.cpp)

 function(add_xpti_library LIB_NAME)