diff --git a/source/common/CMakeLists.txt b/source/common/CMakeLists.txt
index fab88b9e..45766182 100644
--- a/source/common/CMakeLists.txt
+++ b/source/common/CMakeLists.txt
@@ -33,8 +33,7 @@ message(STATUS "Will fetch Unified Memory Framework from ${UMF_REPO}")
 
 include(FetchContent)
 FetchContent_Declare(unified-memory-framework
-    GIT_REPOSITORY    ${UMF_REPO}
-    GIT_TAG           ${UMF_TAG}
+    URL ${UMF_SOURCE_DIR}
 )
 
 if(UR_BUILD_TESTS)
